src/odemis/acq/test/stream_test.py

```diff
@@ -2319,13 +2319,13 @@ def test_acq_arpol(self):
             # check that number of angle resolved images is same as the total number of ebeam positions
             # include if multiple polarization images are required per ebeam position
             ar_das = sas.raw[1:]  # angle resolved data arrays
-            if hasattr(sas, "_polarization") and sas._polarization.value == POL_6POS:
+            if hasattr(sas, "_polarization") and sas._allPol.value:
```

Comment:
I don't think you can make the test case work like this anymore. It used to test all the cases properly because you had `list_positions` containing both the single positions and the "6xPol". Now it will either always have allPol set or never.

So you should change (also) the loop. Maybe loop again with 7 values (add to `list_positions` an extra value), when it is this special value, you'd change `allPol` from False to True.

Also, please avoid accessing the private attributes of the object in the test cases. Technically, you can, but it's bad practice. It's preferable to write something like `if ars.allPol.value:`. (I skip _polarization check explicitly, because we _know_ that this test is checking the polarization, so if the attribute is no there, it should fail (as a sign that one of our assumptions broke).